massa-module-cache/src/controller.rs

```diff
@@ -122,12 +132,20 @@ impl ModuleCache {
         }
     }
 
-    /// Load a cached module for execution and check its validity for execution
+    /// Load a cached module for execution and check its validity for execution.
+    /// Also checks that the provided execution gas is enough to pay for the instance creation cost.
     pub fn load_module(
         &mut self,
         bytecode: &[u8],
         execution_gas: u64,
-    ) -> Result<RuntimeModule, CacheError> {
+    ) -> Result<(RuntimeModule, u64), CacheError> {
```

Comment:
`load_module` ended up not changing anything to the input gas that is later returned but this does still provide multiple things:
* Uniformity with `load_tmp_module` which actually subs costs.
* Simplifies the dev usage after using a cache load. Before the used gas for runtime execution was different in every context, now it's only the return value after loading.
* As module cache functions are always called before any execution (even arbitrary ones) adding pre-execution gas costs becomes a simple addition

Reply:
For ref: https://github.com/massalabs/massa/pull/4505#discussion_r1376135529